src/test/frontend/deploy/deployfromsettings_controller_test.js

```diff
@@ -336,4 +336,39 @@ describe('DeployFromSettings controller', () => {
       expect(ctrl.imagePullSecret).toEqual('');
     });
   });
+
+  /**
+   * Test for the name validation pattern (namePattern)
+   *
+   * The pattern matches strings that start and end with a lowercase letter
+   * and contains only lowercase letters and hyphens.
+   * Full pattern used for validation:
+   *
+   * ^[a-z]+(-[a-z]+)*$
+   */
```

Comment:
Thanks @cheld for digging into this. I've verified that the regex we have here is invalid. Basically we should follow DNS_LABEL spec from here: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L48
Which is slightly different from what we have now. 

@digitalfishpond Can you fix?